ipatests/test_webui/test_user.py

```diff
@@ -376,6 +409,249 @@ def fill_password_dialog(self, password, current=None):
         self.wait_for_request(n=3)
         self.assert_no_error_dialog()
 
+    @screenshot
+    def test_login_without_username(self):
+
+        self.init_app(login='', password='xxx123', skip_username=True)
+
+        alert_e = self.find('.alert[data-name="username"]',
+                            By.CSS_SELECTOR)
+        assert 'Username: Required field' in alert_e.text, 'Alert expected'
+        assert self.login_screen_visible()
+
+    @screenshot
+    def test_disable_delete_admin(self):
+        """
+        Test disabling/deleting admin is not allowed
+        """
+        self.init_app()
+        self.navigate_to_entity('user')
+
+        # try to disable admin user
+        self.select_record('admin')
+        self.facet_button_click('disable')
+        self.dialog_button_click('ok')
+        self.assert_last_error_dialog(ERR_ADMIN_DEL, details=True)
+        self.dialog_button_click('ok')
+        self.assert_record('admin')
+
+        # try to delete admin user. Later we are
+        # confirming by keyboard to test also upstream ticket 4097
```

Comment:
There is no need to use `upstream` here as all ticket references should be upstream unless a full path is provided.